Fix build with boost 1.87 whose math/tools/type_traits.hpp use
C++14 constructs such as is_null_pointer

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -146,7 +146,6 @@ function(check_working_cxx_atomics varname additional_
   include(CheckCXXSourceCompiles)
   include(CMakePushCheckState)
   cmake_push_check_state()
-  set(CMAKE_REQUIRED_FLAGS "-std=c++11")
   set(CMAKE_REQUIRED_LIBRARIES "${additional_lib}")
   set(CMAKE_REQUIRED_QUIET 1)
   CHECK_CXX_SOURCE_COMPILES("
@@ -445,7 +444,6 @@ qt5_add_resources(pulseview_RESOURCES_RCC ${CMAKE_BINA
 add_definitions(-DQT_NO_KEYWORDS)
 add_definitions(-D__STDC_LIMIT_MACROS)
 add_definitions(-Wall -Wextra)
-add_definitions(-std=c++11)
 add_definitions(-DBOOST_MATH_DISABLE_FLOAT128=1)
 
 if(ENABLE_FLOW)
